packages/accounts-password/password_server.js

```diff
@@ -66,7 +66,11 @@ Accounts._checkPassword = function (user, password) {
   password = getPasswordString(password);
 
   if (! bcryptCompare(password, user.services.password.bcrypt)) {
-    result.error = new Meteor.Error(403, "Incorrect password");
+    if(Accounts._options.ambiguousErrorMessages) {
```

Comment:
We should be able to reduce a lot of the error boilerplate code. Creating a re-usable error handling function like the following should help:

```js
const handleError = (msg, throwError = true) => {
  const error = new Meteor.Error(
    403, 
    Accounts._options.ambiguousErrorMessages
      ? "Login failure. Please check your login credentials."
      : msg
  );
  if (throwError) {
    throw error;
  }
  return error;
};
```

So in this case, the call would become:

```
result.error = handleError("Incorrect password", false);
```